packages/nuxt/src/app/nuxt.ts

```diff
@@ -255,7 +255,7 @@ export function createNuxtApp (options: CreateOptions) {
       data: {},
     },
     runWithContext (fn: any) {
-      if (nuxtApp._scope.active) {
+      if (nuxtApp._scope.active && !(getCurrentScope() || getCurrentInstance())) {
         return nuxtApp._scope.run(() => callWithNuxt(nuxtApp, fn))
       }
       return callWithNuxt(nuxtApp, fn)
```

Comment:
I think it's redunant since getCurrentScope is available here, so it will be available later as well until first await. I've checked and it works fine

Reply:
okay, great!

Reply:
But we can probably drop the `getCurrentInstance()` then, right?

Reply:
Agreed.

Reply:
Done!